### PR TITLE
Common db source file menu tree

### PR DESCRIPTION
**Pull request by aminahbl**
Three faceted source tree menu refactor:
- refactor db source menu tree to be reusable in all relevant contexts (originally openned in BuddhaNexus/buddhanexus-frontend-next#187, but pulled into large PR due to branch conflicts)
- refactor db source (previously limit) filter settings to use tree structure (#183)
- add tree breadcrumbs to improve navigation of long file lists

**Note:** this only handles UI refactoring. At the time of writing draft updates to the backend have just become available (details: https://discord.com/channels/1229483442203263056/1229483442677485589/1292608568972804118), and a subsequent PR will update api calls.

### CODE REVIEW

**Comment by sebastian-nehrdich** [comment]
Just a note that the metadata endpoint on the new backend branch which
replaces the sidebar endpoint does have the search_field value so there
should not be any loss in functionality

File: package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "next-mdx-remote": "^4.4.1",
     "next-query-params": "^5.0.0",
     "next-seo": "^6.5.0",
+    "nuqs": "^1.19.3",
```
Comment (XITE-Hubert) [inline package.json:51]: can we remove the other package now?

File: src/components/db/QueryPageTopStack.tsx [hunk 2]
```diff
@@ -39,13 +26,10 @@ export const QueryPageTopStack = ({ matches = 0 }: { matches?: number }) => {
...
   const { fileName, sourceLanguage } = useDbQueryParams();
-  const dbView = useAtomValue(currentViewAtom);
+  const dbView = useAtomValue(currentDbViewAtom);
 
-  const { isSettingsOpen, setIsSettingsOpen } = useSettingsDrawer();
-
-  const [open, setOpen] = React.useState(false);
-  const handleOpen = () => setOpen(true);
-  const handleClose = () => setOpen(false);
+  const setIsSourceTreerOpen = useSetAtom(isDbSourceBrowserDrawerOpenAtom);
```
Comment (XITE-Hubert) [inline src/components/db/QueryPageTopStack.tsx:31]: typo

File: src/components/db/SearchableDbSourceTree/TreeNavigation/TreeBreadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+const handleBreadcrumbClick = ({
+  node,
+  setBreacrumbs,
+}: HandleBreadcrumbClickProps) => {
+  if (!node) return;
+
+  node?.select();
+  node?.toggle();
+
+  const crumbs = getTreeBreadcruumbs(node);
```
Comment (XITE-Hubert) [inline src/components/db/SearchableDbSourceTree/TreeNavigation/TreeBreadcrumbs.tsx:30]: typo

File: src/components/db/SearchableDbSourceTree/TreeNavigation/TreeBreadcrumbs.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  setBreacrumbs(crumbs);
```
Comment (XITE-Hubert) [inline src/components/db/SearchableDbSourceTree/TreeNavigation/TreeBreadcrumbs.tsx:31]: typo

File: src/components/db/SearchableDbSourceTree/utils.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+  return crumbs;
+}
+
+type InitializeTreeProps = {
+  activeTree: TreeApi<DbSourceTreeNode> | null | undefined;
+  setActiveTree: React.Dispatch<
+    React.SetStateAction<TreeApi<DbSourceTreeNode> | null | undefined>
+  >;
+  setBreacrumbs: React.Dispatch<
```
Comment (XITE-Hubert) [inline src/components/db/SearchableDbSourceTree/utils.ts:58]: typo

File: src/features/sidebarSuite/subComponents/settings/DbSourceFilters/styledComponents.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { Box, styled } from "@mui/material";
+
+export const InputOutlineBox = styled(Box)(({ theme }) => ({
+  borderRadius: theme.shape.borderRadius,
+  border: `1px solid ${theme.palette.grey[400]}`,
+  minHeight: "4rem",
+  padding: theme.spacing(1),
+}));
+
+export const ComboboxBox = styled(Box)(({ theme }) => ({
```
Comment (XITE-Hubert) [inline src/features/sidebarSuite/subComponents/settings/DbSourceFilters/styledComponents.tsx:10]: Combobox Box! I like it :)

File: src/features/sidebarSuite/subComponents/settings/DbSourceFilters/tree/DbSourceFilterTree.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    data,
+    height,
+    width,
+    searchTerm,
+    filterSettingName,
+  }: DbSourceTreeBaseProps & Omit<DbSourceFilterSelectorTreeProps, "type">) {
+    const router = useRouter();
+
+    const setActiveTree = useSetAtom(activeDbSourceTreeAtom);
+    const setBreacrumbs = useSetAtom(activeDbSourceTreeBreadcrumbsAtom);
```
Comment (XITE-Hubert) [inline src/features/sidebarSuite/subComponents/settings/DbSourceFilters/tree/DbSourceFilterTree.tsx:32]: typo

**Review by XITE-Hubert — APPROVED** [review]
I only found typos! I can see a lot of work went into this, well done! 👏 

Looks good to merge to me :)

**Comment by XITE-Hubert** [comment]
I did find one bug: 

go into the filters
image

choose e.g. these
image

uncheck "vw" and then "Suttas...". Results in this:
image

**Review by TrebuhD — APPROVED** [review]
I did find one bug: 

go into the filters
image

choose e.g. these
image

lastly, in the input box itself (here):
image

uncheck "vw" and then "Suttas...". Results in this:
image

File: package.json (hunk 1 above)
Comment (aminahbl) [reply package.json:51]: Yes! I was planning to handle this migration in the 2nd PR (one of the reasons I wanted to separate out the work actually.

File: src/components/db/QueryPageTopStack.tsx (hunk 2 above)
Comment (aminahbl) [reply src/components/db/QueryPageTopStack.tsx:31]: ....noooooo "treeer" is a thing! :flushed: :facepalm:

File: src/features/sidebarSuite/subComponents/settings/DbSourceFilters/styledComponents.tsx (hunk 6 above)
Comment (aminahbl) [reply src/features/sidebarSuite/subComponents/settings/DbSourceFilters/styledComponents.tsx:10]: :rofl: This is just the magic I bring! :sparkles:

**Comment by aminahbl** [comment]
@XITE-Hubert Thanks so much for the review, here's specifically what's been updated: https://github.com/BuddhaNexus/buddhanexus-frontend-next/compare/e452b16..029611a

If you're happy please merge and move ahead with the monorepo migration. :pray:

**Pull request merged**